hayeon/baekjoon/수학/2292-벌집.js

```diff
@@ -0,0 +1,36 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const inputData = Number(input[0]);
+  let count = 1;
+  let range = 2;
+
+  if (inputData === 1) {
+    console.log(1);
+  } else {
+    while (range <= inputData) {
+      range = range + 6 * count;
+      count++;
+    }
+    console.log(count);
+  }
```

Comment:
```suggestion
  const inputData = Number(input[0]);
  let count = 1;
  let range = 2;

  while (range <= inputData) {
    range = range + 6 * count++;
  }
  console.log(count);
```
- 하연님 풀이에서 inputData === 1 조건식은 필요없는 코드입니다. 실제로 해당 조건식을 제거하고 1을 넣고 실행해보세요 🙆‍♂️
- 제가 어제 가이드드린 코드에서도 같은 방식으로 리팩토링을 했는데요
```js
range = range + 6 * count;
```
- 위와 같은 코드 더 간략하게 할 수 없을까요?? 🤔
- 추가적으로 증감 연산자를 숫자 뒤에다 붙이면 (++) 현재 값을 활용하고 그 다음 값을 증가시키기 때문에 아래와 같은 형태가 가능합니다
```js
range = range + 6 * count++;
```

Reply:
range += 6 * count; 로 하면 될까요?!?!
와 두번째처럼도 쓸 수 있군요,,,,,? 1 제거하고 한번 돌려보겠습니다!!!

Reply:
맞습니다 :) 기존 값에다 누적시키는 경우에는 +=, -=, *= 연산자들을 활용하면 더욱 간소화된 코드를 짜실 수 있습니다

Reply:
넵 다음부턴 더 신경써서 코드짜겠습니다!!